### PR TITLE
Cf1.7 updates rebase

### PR DESCRIPTION
**Pull request by daltonkell**
A rebased version of #625. Addresses the comments from that PR.

### CODE REVIEW

**Comment by daltonkell** [comment]
Tests are failing because the `test_cdl_nc_file.nc` file no longer exists in the repo (removed because `.nc` files are listed in the `.gitignore`). Options for fixing include:
  - force pushing the file back into the rep
  - removing the test

File: compliance_checker/suite.py [hunk 1]
```diff
@@ -186,6 +186,7 @@ def _get_valid_checkers(self, ds, checker_names):
         the user selected names.
         """
 
+        import pdb; pdb.set_trace()
```
Comment (benjwadams) [inline compliance_checker/suite.py]: Will need to be removed.

Comment (benjwadams) [reply compliance_checker/suite.py]: @daltonkell, I sometimes add a client-side Git pre-commit hook to ensure there are no `pdb` statements present in the code I'm committing.  There are a number of example pre-commit hook scripts out in the wild that might be worth looking at.

**Comment by daltonkell** [comment]
I'll take care of any extraneous code that should be removed (debuggers, comments etc.). Are you comfortable with the added checks? @benjwadams

**Comment by benjwadams** [comment]
Yes, although I may want to put the merge commit off another branch off of master.  I was looking at some of the geoid/vertical datum related checks implemented by CF 1.7 and those will need to be implemented as well before we merge back into the mainline master branch with the CF checker.

**Comment by daltonkell** [comment]
Great, that sounds good to me.

**Comment by benjwadams** [comment]
Please move PR base to branch `cf_1_7_dev`

**Comment by daltonkell** [comment]
Closing in favor of #629

**Pull request closed without merging**